Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/ui/consent/DccReissuanceConsentViewModel.kt

```diff
@@ -63,18 +63,29 @@ class DccReissuanceConsentViewModel @AssistedInject constructor(
     fun navigateBack() = event.postValue(Back)
 
     private suspend fun CertificateReissuance?.toState(): State {
-        var certificate: DccV1.MetaData? = null
-        this?.certificateToReissue?.certificateRef?.barcodeData?.let {
-            certificate = dccQrCodeExtractor.extract(
-                it,
-                DccV1Parser.Mode.CERT_SINGLE_STRICT
-            ).data.certificate
+        val certificates = mutableListOf<DccReissuanceItem>()
+        if (this?.certificates != null) {
+            this.certificates.forEach { dccCertificate ->
+                var certificateToReissue: DccV1.MetaData
+                dccCertificate.certificateToReissue.certificateRef.barcodeData.let {
+                    certificateToReissue = dccQrCodeExtractor.extract(
+                        it,
+                        DccV1Parser.Mode.CERT_SINGLE_STRICT
+                    ).data.certificate
+                }
```

Comment:
```suggestion
                }
```
val certificateToReissue = dccQrCodeExtractor.extract(
                        dccCertificate.certificateToReissue.certificateRef.barcodeData,
                        DccV1Parser.Mode.CERT_SINGLE_STRICT
                    ).data.certificate
          
```suggestion
                }
```